src/linear_algebra/finite_dimensional.lean

```diff
@@ -850,6 +865,46 @@ by { unfold findim, simp [dim_top] }
 
 end top
 
+lemma findim_zero_iff_forall_zero [finite_dimensional K V] :
+  findim K V = 0 ↔ ∀ x : V, x = 0 :=
+begin
+  split; intro h,
+  { haveI := findim_eq_zero_iff_subsingleton.mp h,
+    exact λ x, subsingleton.elim x _ },
+  { rw findim_eq_zero_iff_subsingleton,
+    exact subsingleton_of_forall_eq _ h },
+end
+
+lemma is_basis_of_findim_eq_zero [finite_dimensional K V]
+  {ι : Type*} (h : ¬ nonempty ι) (hV : findim K V = 0) :
+  is_basis K (λ x : ι, (0 : V)) :=
+begin
+  haveI : subsingleton V := findim_eq_zero_iff_subsingleton.1 hV,
+  exact is_basis_empty _ h
+end
+
+lemma is_basis_of_dim_eq_zero {ι : Type*} (h : ¬ nonempty ι)
+  (hV : dim K V = 0) : is_basis K (λ x : ι, (0 : V)) :=
+begin
+  haveI := finite_dimensional_of_dim_eq_zero hV,
+  rw ← findim_eq_dim at hV,
+  norm_cast at hV,
+  exact is_basis_of_findim_eq_zero h hV,
```

Comment:
```suggestion
  haveI : subsingleton V := dim_zero_iff_subsingleton.1 hV,
  exact is_basis_empty _ h
```
which suggests this doesn't belongs in `finite_dimensional` at all, but in `linear_algebra/dimension`